xdsl/irdl/declarative_assembly_format_parser.py

```diff
@@ -187,12 +188,25 @@ def verify_directives(self, elements: list[FormatDirective]):
                     self.raise_error(
                         "A variadic type directive cannot be followed by another variadic type directive."
                     )
-                case VariadicLikeVariable(), VariadicLikeVariable() if not (
-                    isinstance(a, VariadicLikeTypeDirective)
-                    or isinstance(b, VariadicLikeTypeDirective)
-                ):
+                case VariadicLikeVariable(), VariadicLikeVariable():
+                    if not (
+                        isinstance(a, VariadicLikeTypeDirective)
+                        or isinstance(b, VariadicLikeTypeDirective)
+                        or isinstance(a, RegionDirective)
+                        or isinstance(b, RegionDirective)
+                    ):
```

Comment:
Should be equivalent and marginally faster
```suggestion
                    if not (
                        isinstance(a, RegionDirective, VariadicLikeTypeDirective)
                        or isinstance(b, RegionDirective, VariadicLikeTypeDirective)
                    ):
```